sdk/core/core/src/az_http_response_parser.c

```diff
@@ -298,3 +298,33 @@ az_http_response_get_body(az_span const self, az_pair * const p_last_header, az_
   AZ_RETURN_IF_FAILED(az_http_response_parser_get_body(&parser, body));
   return AZ_OK;
 }
+
+/**
+ * Get an HTTP header by name.
+ *
+ * Returns an error if the header is not found.
+ */
+AZ_NODISCARD az_result az_http_response_get_header_by_name(
+    az_span const self,
```

Comment:
The 1st parameter should be our typedef type; not az_span. 'self' should probably be renamed.